### ISSUE TITLE
HTML5 required attribute breaks hack for form submission

### ISSUE THREAD

**Issue by qpleple** [codemirror/CodeMirror-v1#59]
Using the `CodeMirror.fromTextArea()` method, the textarea is hidden and replaced by the CodeMirror element.

Submission is done by the hack:

``` javascript
if (textarea.form) {
  // Deplorable hack to make the submit method do the right thing.
  var rmSubmit = connect(textarea.form, "submit", save, true);
  if (typeof textarea.form.submit == "function") {
    var realSubmit = textarea.form.submit;
    function wrappedSubmit() {
      save();
      textarea.form.submit = realSubmit;
      textarea.form.submit();
      textarea.form.submit = wrappedSubmit;
    }
    textarea.form.submit = wrappedSubmit;
  }
}
```

If the real (and hidden) textarea has the HTML5 `required` attribute, modern browsers check the textarea value before calling `textarea.fom.submit()` (so before "saving" the data) and tries to display an error notice.

The error notice may fail because the textarea is not focusable (it does on my browser, Chrome 15.0).

In short, the form is never submitted and no error is displayed.
